test/module/irohad/multi_sig_transactions/transport_test.cpp

```diff
@@ -87,17 +88,61 @@ TEST(TransportTest, SendAndReceive) {
   // we want to ensure that server side will call onNewState()
   // with same parameters as on the client side
   EXPECT_CALL(*notifications, onNewState(_, _))
-      .WillOnce(Invoke(
-          [&my_key, &cv, &state](const auto &from_key, auto const &target_state) {
-            EXPECT_EQ(my_key.publicKey(), from_key);
+      .WillOnce(Invoke([&my_key, &cv, &state](const auto &from_key,
+                                              auto const &target_state) {
+        EXPECT_EQ(my_key.publicKey(), from_key);
 
-            EXPECT_EQ(state, target_state);
-            cv.notify_one();
-          }));
+        EXPECT_EQ(state, target_state);
+        cv.notify_one();
+      }));
 
   transport->sendState(*peer, state);
   std::unique_lock<std::mutex> lock(mtx);
   cv.wait_for(lock, std::chrono::milliseconds(5000));
 
   server->Shutdown();
 }
+
+/**
+ * @given an instance of MstTransportGrpc initialized using NOT mocked validator
+ * and tx factories dependencies
+ * @when it receives protobuffed MstState that contains a transaction with a
+ * command, where command type is not set
+ * @then there should be no SEGFAULT
+ */
+TEST(TransportTest, TypelessCommand) {
+  auto async_call_ = std::make_shared<
+      iroha::network::AsyncGrpcClient<google::protobuf::Empty>>();
+  std::unique_ptr<shared_model::validation::AbstractValidator<
+      shared_model::interface::Transaction>>
+      tx_validator = std::make_unique<
+          shared_model::validation::
+              DefaultOptionalSignedProtoTransactionValidator>();
```

Comment:
The trouble here is the same as it was in fuzzing pull request - what if the non-mocked validator gets changed in application?

Reply:
I did as @MBoldyrev recommended - reimplemented the test via ITF, thus we always will have the same implementation as in `application` target.